### PR TITLE
uses inplace as op params for GpuGer, GpuGemm, GpuGemmBatch and GpuGemv

### PR DESCRIPTION
**Pull request by tfjgeorge**
this solves #6136 
@notoraptor

### CODE REVIEW

File: theano/gpuarray/blas.py [hunk 1]
```diff
@@ -82,27 +84,24 @@ def perform(self, node, inputs, out_storage):
...
+                 %(out)s = theano_try_copy(%(out)s, %(y)s);
+                 if (%(out)s == NULL) {
+                   %(fail)s
+                 }
+               } else {
+                 Py_XDECREF(%(out)s);
+                 %(out)s = %(y)s;
+                 Py_INCREF(%(out)s);
+               }
+               %(out)s = theano_try_copy(%(out)s, %(y)s);
```
Comment (notoraptor) [inline theano/gpuarray/blas.py]: I think this line (and the 3 ones that follow) are now useless.

File: theano/gpuarray/blas.py [hunk 2]
```diff
@@ -68,9 +70,9 @@ def make_node(self, y, alpha, A, x, beta):
         beta = beta.astype(expected)
         return Apply(self, [y, alpha, A, x, beta], [y.type()])
 
-    def perform(self, node, inputs, out_storage):
+    def perform(self, node, inputs, out_storage, params):
         y, alpha, A, x, beta = inputs
-        inplace = self.inplace
+        inplace = params['inplace']
```
Comment (notoraptor) [inline theano/gpuarray/blas.py]: Use `params.inplace` instead.

File: theano/gpuarray/blas.py [hunk 3]
```diff
@@ -199,45 +199,36 @@ def make_node(self, C, alpha, A, B, beta):
         assert C.ndim == 2
         return Apply(self, [C, alpha, A, B, beta], [C.type()])
 
-    def perform(self, node, inputs, outputs):
+    def perform(self, node, inputs, outputs, params):
         C, alpha, A, B, beta = inputs
-        inplace = self.inplace
+        inplace = params['inplace']
```
Comment (notoraptor) [inline theano/gpuarray/blas.py]: Use `params.inplace` instead.

File: theano/gpuarray/blas.py [hunk 4]
```diff
@@ -282,51 +274,41 @@ def make_node(self, A, alpha, x, y):
         assert y.ndim == 1
         return Apply(self, [A, alpha, x, y], [A.type()])
 
-    def perform(self, node, inp, out):
+    def perform(self, node, inp, out, params):
         A, alpha, x, y = inp
-        inplace = self.inplace
+        inplace = params['inplace']
```
Comment (notoraptor) [inline theano/gpuarray/blas.py]: Use `params.inplace` instead.

**Comment by notoraptor** [comment]
jenkins test this please

**Comment by notoraptor** [comment]
Jenkins test this please

File: theano/gpuarray/blas.py [hunk 5]
```diff
@@ -38,6 +39,7 @@ class GpuGemv(BlasOp):
     Gemv on the GPU.
 
     """
+    params_type = ParamsType(inplace=Scalar('bool'))
```
Comment (abergeron) [inline theano/gpuarray/blas.py]: Probably a better idea to import bool from scalar than to recreate it.  You can import it as `bool_t` so as not to shadow the python bool.

**Comment by abergeron** [comment]
jenkins test this please

**Pull request merged**